Automattic/jetpack
# AI Assistant: Write Brief should not flag words like '2nd' and '100th' as spelling errors.

Partial #39415

## Proposed changes:
Previously spell checker would flag valid number words like "2nd" and "100th" as spelling erorrs:
![Screenshot 2024-10-23 at 10 00 28 AM](https://github.com/user-attachments/assets/4a00dbd4-8079-4c32-92b0-c0c707dda266)

After the fix, we should no longer be flagging those words, but *should* be flagging incorrect words like "32th":
![Screenshot 2024-10-23 at 10 00 59 AM](https://github.com/user-attachments/assets/934fd5f1-14ab-453b-a03e-088d2b9deb61)

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

Create a new post and make sure Write Brief is checking for spelling errors:

![Screenshot 2024-10-23 at 10 20 10 AM](https://github.com/user-attachments/assets/0b58aee8-03ca-4483-a643-40a24e06fb74)

Add some text that includes number words. Here's the test post I have been using:

> These should be good: 10th and 1st and 2nd and 13th and 11th and 21st and 99th and 20th and 33rd and 19th and 1000129th and 939483rd 999,999th
> 
> Some wrong ones: 3th, 5nd, 238st, 1,000rd 100,823nd (kinda weird since it splits on the comma, but still fine
> 
> I guess spell checker is fine with 3rd and 1st but not 10th or 7th or 2195th but 3492nd is also not fine, neither is 2nd. 
> 
> Since the regex is now ingesting numbers, we also need to validate that raw numbers are valid: 1023 99 12,202,284 5555555 1 22 31 5 99 999

Verify that words like "10th", "13th", and "2nd" are being flagged as spelling mistakes (usually just the non-numeric "th" is underlined).

Check out this branch and rebuild Jetpack.

Reload the post and observe that all of the previously flagged words are no longer underlined. Instead the incorrect words such as "3th", and "238st" should be underlined in red.

Also validate that none of the raw numbers (without a "th" etc at the end) are being flagged as spelling errors.
